ord_schema/units.py

```diff
@@ -237,17 +287,109 @@ def resolve(self, string: str, allow_range: bool = False) -> ord_schema.UnitMess
         else:
             value = float(value)
         assert string_unit is not None  # Type hint.
+        message, unit = self.resolve_unit(string_unit)
+        if value < 0.0 and message != reaction_pb2.Temperature:
+            raise ValueError(f"negative values are only allowed for temperature: {string}")
+        if precision:
+            return message(value=value, precision=precision, units=unit)
+        return message(value=value, units=unit)
+
+    def resolve_unit(self, string_unit: str) -> Tuple[ord_schema.UnitMessage, int]:
+        """Resolves a unit string into its message type and unit ENUM value.
+
+        Args:
+            string_unit: The string unit to parse; for example: "gram".
+
+        Returns:
+            Tuple containing the message type and unit ENUM value.
+
+        Raises:
+            KeyError: if string unit cannot be parsed.
+        """
         string_unit = string_unit.lower()
         if string_unit in self._forbidden_units:
             raise KeyError(f"forbidden units: {string_unit}: " f"({self._forbidden_units[string_unit]})")
         if string_unit not in self._resolver:
             raise KeyError(f"unrecognized units: {string_unit}")
         message, unit = self._resolver[string_unit]
-        if value < 0.0 and message != reaction_pb2.Temperature:
-            raise ValueError(f"negative values are only allowed for temperature: {string}")
-        if precision:
-            return message(value=value, precision=precision, units=unit)
-        return message(value=value, units=unit)
+        return (message, unit)
+
+    def convert(self, message: ord_schema.UnitMessage, new_units: Union[str, int]) -> ord_schema.UnitMessage:
+        """Converts a united message into another united message of the same
+        type, but with different units.
+
+        Args:
+            message: a message with units, e.g., Mass, Length.
+            new_units: the desired units of the new message, expressed
+                either as a string or an integer (ENUM value). Use of a
+                string is recommended due to the ambiguity of using
+                ENUM values; for example, Mass.GRAM == Time.MINUTE.
+
+        Returns:
+            A new message with units, e.g., Mass, Length.
+        """
+        #pylint: disable=too-many-branches
```

Comment:
looks like the missing space after `#` is causing errors in format tests (GitHub actions)

Reply:
Thanks for the catch